### PR TITLE
update opentelemetry dependencies

### PR DESCRIPTION
**Pull request by jinja2**
**Description:** <Describe what has changed.>

- Update Opentelemetry dependencies to latest. 
- Updates mdatagen package to `go.opentelemetry.io/collector/cmd/mdatagen`.
- Remove the disabled feature gate `telemetry.useOtelForInternalMetrics` which is now enabled by default.
- Updates default configs to drop new internal metrics exposed by the collector. These are metrics tracking grpc and http request/responses.
- Updates to tests for above changes

**Link to Splunk idea:** <Link to Splunk idea, see https://ideas.splunk.com>

**Testing:** <Describe what testing was performed and which tests were added.>

**Documentation:** <Describe the documentation added.>

### CODE REVIEW

**Review by dmitryax — APPROVED** [review]

File: tests/receivers/prometheus/testdata/resource_metrics/internal.yaml [hunk 1]
```diff
@@ -4,6 +4,9 @@ resource_metrics:
       net.host.port: "8889"
       service.instance.id: localhost:8889
       service.name: otel-agent
+      service_instance_id: <ANY>
+      service_name: otelcol
+      service_version: <VERSION_FROM_BUILD>
```
Comment (jinja2) [inline tests/receivers/prometheus/testdata/resource_metrics/internal.yaml:9]: These are added to resource attributes due to `target_info` metric now being exposed. The prometheus receiver adds the [labels on target_info to resource attrs](https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/main/receiver/prometheusreceiver/internal/transaction.go#L323).

Comment (dmitryax) [reply tests/receivers/prometheus/testdata/resource_metrics/internal.yaml:9]: I think that's ok. Ideally this should've been configurable on the Prometheus receiver side

Comment (jinja2) [reply tests/receivers/prometheus/testdata/resource_metrics/internal.yaml:9]: Yes, just wanted to note that all internal metrics have 2 sets of `service name`, and `service instance id`. The [ones added by the receiver](https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/main/receiver/prometheusreceiver/internal/prom_to_otlp.go#L42-L48) in otel format and a 2nd set from target_info with different values, which might become confusing.

File: tests/receivers/discovery/testdata/host_observer_simple_prometheus_config.yaml [hunk 2]
```diff
@@ -13,14 +13,14 @@ receivers:
     embed_receiver_config: true
     receivers:
       prometheus_simple:
-        rule: type == "hostport"
+        rule: type == "hostport" and command contains "otelcol"
```
Comment (jinja2) [inline tests/receivers/discovery/testdata/host_observer_simple_prometheus_config.yaml:16]: Without this change in rule, too many ports were being discovered and the test would timeout before it even tried the internal metrics port. I also updated the metric to match against for a successful status to get more reliable runs.

File: tests/receivers/prometheus/testdata/resource_metrics/internal.yaml (hunk 1 above)
Comment (dmitryax) [reply tests/receivers/prometheus/testdata/resource_metrics/internal.yaml:9]: @jinja2 do they have different keys with the same values?

Comment (jinja2) [reply tests/receivers/prometheus/testdata/resource_metrics/internal.yaml:9]: Are you asking about attributes from target_info and the ones added by the receiver, right? I don't think I have seen any in the tests I have worked on.

**Pull request merged**